upstairs/src/downstairs.rs

```diff
@@ -3722,562 +3672,196 @@ impl Downstairs {
         }
     }
 
-    #[cfg(feature = "notify-nexus")]
-    fn get_target_addrs(&self) -> Vec<SocketAddr> {
-        self.clients
-            .iter()
-            .filter_map(|client| client.target_addr)
-            .collect()
-    }
-
-    #[cfg(feature = "notify-nexus")]
-    fn notify_nexus_of_live_repair_start(&self, repair: &LiveRepairData) {
-        let log = self.log.new(o!("repair" => repair.id.to_string()));
-
-        let mut repairs = Vec::with_capacity(repair.repair_downstairs.len());
-
-        for cid in &repair.repair_downstairs {
-            let Some(region_uuid) = self.clients[*cid].id() else {
-                // A downstairs doesn't have an id but is being repaired...?
-                warn!(log, "downstairs {cid} has a None id?");
-                continue;
-            };
+    fn notify_live_repair_start(&self, repair: &LiveRepairData) {
+        if let Some(notify) = &self.notify {
+            let log = self.log.new(o!("repair" => repair.id.to_string()));
+            let mut repairs =
+                Vec::with_capacity(repair.repair_downstairs.len());
 
-            let Some(target_addr) = self.clients[*cid].target_addr else {
-                // A downstairs doesn't have a target_addr but is being
-                // repaired...?
-                warn!(log, "downstairs {cid} has a None target_addr?");
-                continue;
-            };
+            for cid in &repair.repair_downstairs {
+                let Some(region_uuid) = self.clients[*cid].id() else {
+                    // A downstairs doesn't have an id but is being repaired...?
+                    warn!(log, "downstairs {cid} has a None id?");
+                    continue;
+                };
 
-            repairs.push(DownstairsUnderRepair {
```

Comment:
It's a bit of a shame to change this struct into a tuple, but I wouldn't block the PR for it

Reply:
Agreed, but it _is_ intentional; I wanted to remove Nexus-flavored types from every module except `notify.rs`.  We could add a new intermediate-type, but probably not worth it...